core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/EnvelopePhysics.java

```diff
@@ -43,7 +44,7 @@ public static double interpolateStepTime(
             double positionDelta
     ) {
         var acceleration = stepAcceleration(lastPos, nextPos, lastSpeed, nextSpeed);
-        if (acceleration == 0.0)
+        if (areSpeedsEqual(acceleration, 0.0))
```

Comment:
Add an `areAccelerationsEqual` with a corresponding acceleration epsilon, as it should be lower than the speed epsilon.

Reply:
I've added a new function to keep the right semantic, but the epsilon can't be much lower, because we divide by the result any small value would be problematic.
We really should get rid of floats as soon as possible